### PR TITLE
Cleans up TPM format zero error messages.

### PR DESCRIPTION
**Pull request by Superhepper**
- Changes the error messages so that they form proper sentences.

- Removes comments from the error messages (e.g. text that contains the work 'NOTE').

- Adds test for the display trait implementation.

### CODE REVIEW

**Comment by Superhepper** [comment]
Could you please check the sentences. Because sometimes it seems to me as if the descriptions in the specification are not proper sentences and rather sloppy written.

File: tss-esapi/src/context/tpm_commands/testing.rs [hunk 1]
```diff
@@ -33,7 +33,11 @@ impl Context {
 
     /// Get the TPM self test result
     ///
+    /// # Details
     /// The returned buffer data is manufacturer-specific information.
+    ///
+    /// TpmFormatZeroResponseCode::Error(TpmFormatZeroError::Failure) may be returned by TPM2_GetTestResult()
+    /// as the 'testResultparameter'.
```
Comment (wiktor-k) [inline tss-esapi/src/context/tpm_commands/testing.rs:40]: :thinking: to be honest this is hard to parse but frankly, the function signature doesn't look nice too: `Result<(MaxBuffer, Result<()>)>`.

Maybe it'd be a good idea to linkify `TpmFormatZeroResponseCode::Error(TpmFormatZeroError::Failure)` (or at least add backticks).

Why mention `TPM2_GetTestResult()` here? Because it's an underlying routine? Maybe using `get_test_result` would be better?

What's `'testResultparameter'`?

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs [hunk 2]
```diff
@@ -5,6 +5,14 @@ use std::convert::TryFrom;
...
 /// Type representing the TPM format zero error
 /// response code.
+///
+/// # Details
+/// The error messages are short forms of the descriptions given in the specification
+/// that describes return codes (see the
+/// [Part 2, Structures](https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part2_Structures_pub.pdf)).
+/// Sometimes these descriptions refers to fields within the structures described in the specification. When
+/// a message contains such a description then the name of the of the field is put inside single quotation marks
+/// (e.g. 'authValue').
```
Comment (wiktor-k) [inline tss-esapi/src/error/return_code/tpm/format_zero/error.rs:15]: I think using backticks for `authValue` would be slightly better.

File: tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_zero_warning_tests.rs [hunk 3]
```diff
@@ -74,7 +74,7 @@ macro_rules! test_display_trait_impl {
                 TpmFormatZeroWarningResponseCode::from(TpmFormatZeroWarning::$zero_warning)
             ),
             $expected_error_message,
-            "BaseReturnCode with {} did not produce the expected error message",
+            "TpmFormatZeroWarningResponseCode with {} did not produce the expected error message",
```
Comment (wiktor-k) [inline tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_zero_warning_tests.rs:77]: Just [recently learned](https://github.com/wiktor-k/testresult/pull/3/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R18) that [`std::any::type_name::<T>()`](https://doc.rust-lang.org/std/any/fn.type_name.html) can be used to convert type name to string at compile-time.

(Not necessarily suggesting this change here, just noting that this would be compile-time checked and would survive type rename).

**Review by wiktor-k — APPROVED** [review]
Looks good :+1: Just have a couple of issues (maybe with my understanding ;) ).

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs [hunk 4]
```diff
@@ -50,40 +58,40 @@ impl std::error::Error for TpmFormatZeroErrorResponseCode {}
...
+            TpmFormatZeroError::NvSize => write!(f, "Requested allocation size is larger than allowed."),
+            TpmFormatZeroError::NvLocked => write!(f, "NV access locked."),
+            TpmFormatZeroError::NvAuthorization => write!(f, "NV access authorization fails in command actions."),
+            TpmFormatZeroError::NvUninitialized => write!(f, "An NV Index is used before being initialized or the state saved by TPM2_Shutdown(STATE) could not be restored."),
+            TpmFormatZeroError::NvSpace => write!(f, "Insufficient space for NV allocation."),
+            TpmFormatZeroError::NvDefined => write!(f, "NV Index or persistent object already defined."),
+            TpmFormatZeroError::BadContext => write!(f, "Context in TPM2_ContextLoad() is not valid."),
+            TpmFormatZeroError::CpHash => write!(f, "'cpHash' value already set or not correct for use."),
+            TpmFormatZeroError::Parent => write!(f, "Handle for parent is not a valid parent."),
+            TpmFormatZeroError::NeedsTest => write!(f, "Some function needs testing."),
```
Comment (wiktor-k) [inline tss-esapi/src/error/return_code/tpm/format_zero/error.rs:92]: This looks a little bit off. For me even `Function needs testing` is OK.

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs [hunk 5]
```diff
@@ -50,40 +58,40 @@ impl std::error::Error for TpmFormatZeroErrorResponseCode {}
...
+            TpmFormatZeroError::NoResult => write!(f, "Returned when an internal function cannot process a request due to an unspecified problem. This code is usually related to invalid parameters that are not properly filtered by the input unmarshaling code."),
```
Comment (wiktor-k) [inline tss-esapi/src/error/return_code/tpm/format_zero/error.rs:93]: I think `Returned when an` can be dropped so that it starts with `Internal function cannot process a request due to an unspecified problem`

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs [hunk 6]
```diff
@@ -50,40 +58,40 @@ impl std::error::Error for TpmFormatZeroErrorResponseCode {}
...
-            TpmFormatZeroError::Parent => write!(f, "handle for parent is not a valid parent"),
-            TpmFormatZeroError::NeedsTest => write!(f, "some function needs testing."),
-            TpmFormatZeroError::NoResult => write!(f, "returned when an internal function cannot process a request due to an unspecified problem. This code is usually related to invalid parameters that are not properly filtered by the input unmarshaling code."),
-            TpmFormatZeroError::Sensitive => write!(f, "the sensitive area did not unmarshal correctly after decryption - this code is used in lieu of the other unmarshaling errors so that an attacker cannot determine where the unmarshaling error occurred"),
+            TpmFormatZeroError::Initialize => write!(f, "TPM not initialized by TPM2_Startup or already initialized."),
+            TpmFormatZeroError::Failure => write!(f, "Commands not being accepted because of a TPM failure."),
+            TpmFormatZeroError::Sequence => write!(f, "Improper use of a sequence handle."),
+            TpmFormatZeroError::Private => write!(f, "Not currently used."),
+            TpmFormatZeroError::Hmac => write!(f, "Not currently used."),
```
Comment (wiktor-k) [inline tss-esapi/src/error/return_code/tpm/format_zero/error.rs:65]: These two can be folded `TpmFormatZeroError::Private | TpmFormatZeroError::Hmac`? Not sure if it improves readability that much though :sweat_smile:

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs [hunk 7]
```diff
@@ -50,40 +58,40 @@ impl std::error::Error for TpmFormatZeroErrorResponseCode {}
...
-            TpmFormatZeroError::BadContext => write!(f, "context in TPM2_ContextLoad() is not valid"),
-            TpmFormatZeroError::CpHash => write!(f, "cpHash value already set or not correct for use"),
-            TpmFormatZeroError::Parent => write!(f, "handle for parent is not a valid parent"),
-            TpmFormatZeroError::NeedsTest => write!(f, "some function needs testing."),
-            TpmFormatZeroError::NoResult => write!(f, "returned when an internal function cannot process a request due to an unspecified problem. This code is usually related to invalid parameters that are not properly filtered by the input unmarshaling code."),
-            TpmFormatZeroError::Sensitive => write!(f, "the sensitive area did not unmarshal correctly after decryption - this code is used in lieu of the other unmarshaling errors so that an attacker cannot determine where the unmarshaling error occurred"),
+            TpmFormatZeroError::Initialize => write!(f, "TPM not initialized by TPM2_Startup or already initialized."),
+            TpmFormatZeroError::Failure => write!(f, "Commands not being accepted because of a TPM failure."),
```
Comment (wiktor-k) [inline tss-esapi/src/error/return_code/tpm/format_zero/error.rs:62]: `Commands not accepted because of a TPM failure` :thinking:

File: tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_zero_warning_tests.rs (hunk 3 above)
Comment (ionut-arm) [reply tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_zero_warning_tests.rs:77]: That's pretty cool, thanks for sharing!

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs (hunk 6 above)
Comment (Superhepper) [reply tss-esapi/src/error/return_code/tpm/format_zero/error.rs:65]: Nah and besides I think they are just place holders for something that might or might not come.

File: tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_zero_warning_tests.rs (hunk 3 above)
Comment (Superhepper) [reply tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_zero_warning_tests.rs:77]: Nice I made use of it. Because it also gives the whole module path not just the type name.

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs (hunk 5 above)
Comment (Superhepper) [reply tss-esapi/src/error/return_code/tpm/format_zero/error.rs]: Fixed.

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs (hunk 7 above)
Comment (Superhepper) [reply tss-esapi/src/error/return_code/tpm/format_zero/error.rs]: Fixed.

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs (hunk 4 above)
Comment (Superhepper) [reply tss-esapi/src/error/return_code/tpm/format_zero/error.rs]: Fixed.

File: tss-esapi/src/error/return_code/tpm/format_zero/error.rs (hunk 2 above)
Comment (Superhepper) [reply tss-esapi/src/error/return_code/tpm/format_zero/error.rs]: Fixed.

File: tss-esapi/src/context/tpm_commands/testing.rs (hunk 1 above)
Comment (Superhepper) [reply tss-esapi/src/context/tpm_commands/testing.rs]: Fixed but please check again if the new version is better.

Comment (wiktor-k) [reply tss-esapi/src/context/tpm_commands/testing.rs]: I think it looks very nice :ok_hand: thanks!

File: tss-esapi/src/context/tpm_commands/testing.rs [hunk 8]
```diff
@@ -33,7 +33,28 @@ impl Context {
...
+    ///
+    /// - A [TpmFormatZeroErrorResponseCode](crate::error::TpmFormatZeroErrorResponseCode) with a `NeedsTest`
+    ///   [TpmFormatZeroError](crate::constants::return_code::TpmFormatZeroError) indicates that no self test
+    ///   has been performed and testable function has not been tested.
+    ///
+    /// - A [TpmFormatZeroErrorResponseCode](crate::error::TpmFormatZeroErrorResponseCode) with a `Failure`
+    ///   [TpmFormatZeroError](crate::constants::return_code::TpmFormatZeroError) indicates that there was
+    ///   a failure.
+    ///
+    /// See [Part 3, Commands](https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part3_Commands_pub.pdf).
```
Comment (ionut-arm) [inline tss-esapi/src/context/tpm_commands/testing.rs:57]: Very comprehensive, thanks!

**Review by ionut-arm — APPROVED** [review]
🙏🏻 thanks for the patch!
